hack/gen-docs.sh

```diff
@@ -0,0 +1,76 @@
+#	   'master' docs directory, updating any version-specific links (e.g. to a
+#	   specific branch of the GitHub repository) to use the new version.
+#
+# The unstaged changes in the working directory can now be diff'ed against the
+# staged changes using 'git diff' to review all docs changes made since the previous 
+# tagged version. Once the unstaged changes are ready, they can be added to the
+# staging area using 'git add' and then committed.
+#
+# NOTE: this script is just one step in fully generating a new set of versioned
+# docs. The full process is documented in site/README-JEKYLL.md.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# don't run if there's already a directory for the target docs version
+if [[ -d site/docs/$NEW_DOCS_VERSION ]]; then
+    echo "ERROR: site/docs/$NEW_DOCS_VERSION already exists"
+    exit 1
+fi
+
+# get the alphabetically last item in site/docs to use as PREVIOUS_DOCS_VERSION
+# if not explicitly specified by the user
+if [[ -z "${PREVIOUS_DOCS_VERSION:-}" ]]; then
+    echo "PREVIOUS_DOCS_VERSION was not specified, getting the latest version"
+    PREVIOUS_DOCS_VERSION=$(ls -1 site/docs/ | tail -n 1)
+fi
+
+# make a copy of the previous versioned docs
+echo "Creating copy of docs directory site/docs/$PREVIOUS_DOCS_VERSION in site/docs/$NEW_DOCS_VERSION"
+cp -r site/docs/${PREVIOUS_DOCS_VERSION}/ site/docs/${NEW_DOCS_VERSION}/
+
+# 'git add' the previous version's docs as-is so we get a useful diff when we copy the master docs in
+echo "Running 'git add' for previous version's doc contents to use as a base for diff"
+git add site/docs/${NEW_DOCS_VERSION}
+
+
+# now copy the contents of site/docs/master into the same directory so we can get a nice
+# git diff of what changed since previous version
+rm -rf site/docs/${NEW_DOCS_VERSION}/ && cp -r site/docs/master/ site/docs/${NEW_DOCS_VERSION}/
+
+# replace known version-specific links -- the sed syntax is slightly different in OS X and Linux,
+# so check which OS we're running on.
+if [[ $(uname) == "Darwin" ]]; then
+    echo "[OS X] updating version-specific links"
+    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://velero.io/docs/master|https://velero.io/docs/$NEW_DOCS_VERSION|g"
+    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://github.com/heptio/velero/blob/master|https://github.com/heptio/velero/blob/$NEW_DOCS_VERSION|g"
+else
+    echo "[Linux] updating version-specific links"
+    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i'' "s|https://velero.io/docs/master|https://velero.io/docs/$NEW_DOCS_VERSION|g"
+    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i'' "s|https://github.com/heptio/velero/blob/master|https://github.com/heptio/velero/blob/$NEW_DOCS_VERSION|g"
+fi
+
+echo "Success! site/docs/$NEW_DOCS_VERSION has been created. You can now run 'git diff' to review all docs changes made since the previous tagged version."
```

Comment:
May be worth referring to the manual steps in README-JEKYLL.md in this message so they're less likely to be missed.